handler/src/main/java/io/netty/handler/ssl/SslHandler.java

```diff
@@ -494,6 +494,12 @@ public void flush(ChannelHandlerContext ctx) throws Exception {
         }
         try {
             wrap(ctx, false);
+        } catch (Exception e) {
+            // Just fail the pending writes in case of an SSLException. If there are no pending writes rethrow.
+            if (pendingUnencryptedWrites.removeAndFailAll(e)) {
```

Comment:
I am still not sure I understand what "use-case" you are talking about... Basically I think if we have pending writes we should just fail them as wrap(...) is called by flush(). If we have no pending writes we should propagate as otherwise we will just suppress an exception which is never good. That said imho wrap() should not really throw any exceptions if the queue is empty but better safe then sorry.